src/main/java/teammates/ui/controller/Action.java

```diff
@@ -120,7 +125,7 @@ public void setEmailSender(EmailSender emailSender) {
     }
     
     protected void authenticateUser() {
-        UserType currentUser = gateKeeper.getCurrentUser();
+        UserType currentUser = logic.getCurrentUser();
```

Comment:
Well, it appears as gateKeeper on my updated master. You must have accidentally changed it while refactoring.